### PR TITLE
add foldM

### LINKED ISSUE #65: foldM?
Seems there are specialized versions for `List` and `Array`, but no generic version - is this intentional?

### PR DESCRIPTION
**Pull request by clayrat**
Fixes #65

### CODE REVIEW

File: src/Data/Foldable.purs [hunk 1]
```diff
@@ -169,6 +170,10 @@ instance foldableMultiplicative :: Foldable Multiplicative where
 fold :: forall f m. Foldable f => Monoid m => f m -> m
 fold = foldMap id
 
+-- | Similar to 'foldl', but the result is encapsulated in a monad.
```
Comment (paf31) [inline src/Data/Foldable.purs]: We should add a note that this isn't stack-safe in general, for any monad which involves building up a thunk, like `Eff`.

Comment (paf31) [inline src/Data/Foldable.purs]: Of course, it could be stack-safe, if it were implemented using `List.fromFoldable`, but that's a different approach.

Comment (paf31) [inline src/Data/Foldable.purs]: Any thoughts on this @garyb?

Comment (garyb) [inline src/Data/Foldable.purs]: I think it's fine as-is, with an updated comment. Maybe we could provide a `foldMRec` too later, although I'm not sure that's even implementable with the same signature.

Comment (paf31) [inline src/Data/Foldable.purs]: @clayrat Could you please update the comment? Thanks!

Comment (clayrat) [inline src/Data/Foldable.purs]: @paf31 done

Comment (paf31) [inline src/Data/Foldable.purs]: Thanks!

**Pull request merged**